src/BuildIt.Forms/BuildIt.Forms.Controls/Platforms/Android/AutoFitTextureView.cs

```diff
@@ -45,17 +48,45 @@ protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
             if (ratioWidth == 0 || ratioHeight == 0)
             {
                 SetMeasuredDimension(width, height);
+                return;
             }
-            else
+
+            switch (aspect)
             {
-                if (width < (float)height * ratioWidth / (float)ratioHeight)
-                {
-                    SetMeasuredDimension(width, width * ratioHeight / ratioWidth);
-                }
-                else
-                {
-                    SetMeasuredDimension(height * ratioWidth / ratioHeight, height);
-                }
+                case Aspect.AspectFit:
+                    if (width < (float)height * ratioWidth / (float)ratioHeight)
+                    {
+                        SetMeasuredDimension(width, width * ratioHeight / ratioWidth);
+                    }
+                    else
+                    {
+                        SetMeasuredDimension(height * ratioWidth / ratioHeight, height);
+                    }
+
+                    break;
+
+                case Aspect.AspectFill:
+                    var previewAspectRatio = (double)ratioWidth / ratioHeight;
```

Comment:
When casting, we should be consistent as to whether we use float or double

Reply:
yep will change